core/src/main/scala/org/apache/spark/util/io/ChunkedByteBufferOutputStream.scala

```diff
@@ -30,9 +31,14 @@ import org.apache.spark.storage.StorageUtils
  * @param chunkSize size of each chunk, in bytes.
  */
 private[spark] class ChunkedByteBufferOutputStream(
-    chunkSize: Int,
+    var chunkSize: Int,
     allocator: Int => ByteBuffer)
-  extends OutputStream {
+  extends OutputStream with Logging{
+
+  if (chunkSize < 0) {
+    logWarning(s"chunkSize should not be an negative value, replaced as 4MB default.")
+    chunkSize = 4 * 1024 * 1024
+  }
 
```

Comment:
For now though, there are not too may call sites. Checking Long value is not the best but feasible way to address this problem.